src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -139,6 +149,16 @@ function assertValidProps(component, props) {
     'not a string. For example, style={{marginRight: spacing + \'em\'}} when ' +
     'using JSX.'
   );
+  if (__DEV__) {
+    Object.keys(props).forEach(function(propName) {
+      warning(
+        !eventTypes.hasOwnProperty(propName),
```

Comment:
good catch

how about building a regexp with the case insensitive flag when creating the map, and test anything matching the pattern, but not equal to the valid prop?